pkg/operator/configobserver/apiserver/observe_cors_test.go

```diff
@@ -42,36 +43,50 @@ func TestObserveAdditionalCORSAllowedOrigins(t *testing.T) {
 		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
-			if tt.corsConfig != nil {
-				if err := indexer.Add(&configv1.APIServer{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "cluster",
-					},
-					Spec: configv1.APIServerSpec{
-						AdditionalCORSAllowedOrigins: tt.corsConfig,
-					},
-				}); err != nil {
-					t.Fatal(err)
-				}
-			}
-			listers := testLister{
-				lister: configlistersv1.NewAPIServerLister(indexer),
+		for _, useAPIServerArguments := range []bool{false, true} {
+			var corsPath []string
+			if useAPIServerArguments {
+				corsPath = []string{"apiServerArguments", "cors-allowed-origins"}
+			} else {
+				corsPath = []string{"corsAllowedOrigins"}
 			}
+			t.Run(tt.name, func(t *testing.T) {
+				indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+				if tt.corsConfig != nil {
+					if err := indexer.Add(&configv1.APIServer{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "cluster",
+						},
+						Spec: configv1.APIServerSpec{
+							AdditionalCORSAllowedOrigins: tt.corsConfig,
+						},
+					}); err != nil {
+						t.Fatal(err)
+					}
+				}
+				listers := testLister{
+					lister: configlistersv1.NewAPIServerLister(indexer),
+				}
 
-			gotConfig, errs := ObserveAdditionalCORSAllowedOrigins(listers, events.NewInMemoryRecorder(t.Name()), tt.existingConfig)
-			if len(errs) > 0 {
-				t.Errorf("ObserveAdditionalCORSAllowedOrigins() expected no errors, got %v", errs)
-			}
+				var gotConfig map[string]interface{}
+				var errs []error
+				if useAPIServerArguments {
+					gotConfig, errs = ObserveAdditionalCORSAllowedOriginsToArguments(listers, events.NewInMemoryRecorder(t.Name()), tt.existingConfig)
+				} else {
+					gotConfig, errs = ObserveAdditionalCORSAllowedOrigins(listers, events.NewInMemoryRecorder(t.Name()), tt.existingConfig)
+				}
+				if len(errs) > 0 {
+					t.Errorf("ObserveAdditionalCORSAllowedOrigins() expected no errors, got %v", errs)
+				}
 
-			gotCors := gotConfig["corsAllowedOrigins"].([]interface{})
-			for i := range tt.expectedCORS {
-				if gotCors[i] != tt.expectedCORS[i] {
-					t.Fatalf("ObserveAdditionalCORSAllowedOrigins() got = %v, want %v", gotCors, tt.expectedCORS)
+				gotCors, _, _ := unstructured.NestedStringSlice(gotConfig, corsPath...)
```

Comment:
okay, this actually checks the `paths` so we now that the config is in good shape.